src/Makefile.am

```diff
@@ -74,16 +74,26 @@ libfapolicyd_la_SOURCES += \
 
 endif
 
+if NEED_MD5
+libfapolicyd_la_SOURCES += \
+	library/md5-backend.c \
+	library/md5-backend.h
+endif
+
 if WITH_DEB
-libfapolicyd_la_SOURCES += library/deb-backend.c
+libfapolicyd_la_SOURCES += library/deb-backend.c \
+	library/md5-backend.c \
+	library/md5-backend.h
 fapolicyd_CFLAGS += -DLIBDPKG_VOLATILE_API
 fapolicyd_LDFLAGS += -ldpkg
 endif
 
-if NEED_MD5
-libfapolicyd_la_SOURCES += \
+if WITH_EBUILD
+libfapolicyd_la_SOURCES += library/ebuild-backend.c \
 	library/md5-backend.c \
-	library/md5-backend.h
+	library/md5-backend.h \
```

Comment:
Since `WITH_EBUILD` pulls in `NEED_MD5` now, this should be removable.  I would similarly suggest that `WITH_DEB` should be checked to ensure it does, and remove that from the `WITH_DEB` chunk as well.